iroh/src/node.rs

```diff
@@ -567,15 +581,21 @@ impl RpcHandler {
     ) -> impl Stream<Item = ListCollectionsResponse> + Send + 'static {
         // collections are always stored internally, so we take everything that is stored internally
         // and try to parse it as a collection
-        let items = self.inner.db.internal().filter_map(|(hash, collection)| {
-            Collection::from_bytes(&collection)
-                .ok()
-                .map(|collection| ListCollectionsResponse {
-                    hash,
-                    total_blobs_count: collection.blobs().len(),
-                    total_blobs_size: collection.total_blobs_size(),
+        let items = if let Some(db) = self.concrete_db() {
```

Comment:
Why can't I list collections with a generic `BaoCollection`?

Reply:
You can list blobs, but what even is a collection in a blob store? Everything that has been explicitly added as a collection, or things that were added as blobs and just happen to deserialize as a collection? And what if you happen to change what a collection is using the custom collection stuff?

Once the traits are a bit more fleshed out, you would have to separately track the collections as the "gc roots". But this really doesn't have much to do with what BaoCollection (whatever we end up naming it) is supposed to be - just a thing that you can use to serve requests.

I guess I could have an additional trait containing stuff we also want to have generic but aren't quite as sure about yet. Maybe I will do that.

Reply:
I think the confusing piece here is that `Collections` will not be listed under `list_blobs` for `Database` but they would be for `BaoCollection`. So I would really like to find a more consistent approach if we add this abstraction level.

Reply:
I changed it now so that you have an additional trait that adds blobs and roots fns. blobs lists all blobs including collections.